src/main/resources/workerScripts/fromMultiplePriorityQueues.lua

```diff
@@ -16,7 +18,7 @@ for q in queues.gmatch(queues, NEXT_QUEUE_REGEX) do
     local status, queueType = next(redis.call('TYPE', queueName))
     local payload
     if queueType == 'zset' then
-        local firstMsg = redis.call('ZRANGE', queueName, '0', '0')
+	    local firstMsg = redis.call('ZRANGEBYSCORE', queueName, '-inf', now, 'WITHSCORES', 'LIMIT' , '0' , '1')
```

Comment:
@dvizelman thanks for this details explanation!
@talhasson This sounds like our Cassandra time based filtering code code, playing with time can be dangerous :smiling_imp: